src/scmdata/run.py

```diff
@@ -988,6 +988,58 @@ def _meta_column(self, col) -> pd.Series:
         out = self._meta.get_level_values(col)
         return pd.Series(out, name=col, index=self._df.columns)
 
+    def set_meta(self, dimension: str, value: Any, **filter_kwargs: Any):
+        """
+        Update metadata of timeseries
+
+        Optionally, a subset of timeseries may be modified through the use of
+        additional `filter_kwargs` which are passed to :func:`filter`. The meta
+        for the non-filtered timeseries are not modified.
+
+        This method does not preserve the order of the timeseries.
+
+        Parameters
+        ----------
+        dimension : str
+            Dimension of meta to update
+
+        value : Any
+            Value to set the targeted meta to
+
+        filter_kwargs : Any
+            Arguments used to filter which timeseries are updated
+
+            All the filtering functionality of :func:`filter` is available, except for
+            `"inplace"`.
+
+        See Also
+        --------
+        :func:`filter`
+
+        Returns
+        -------
+        :class:`BaseScmRun <scmdata.run.BaseScmRun>`
```

Comment:
Can you remind me again why we use class not obj and why we need the < and > ? I remember you explaining but I can't remember

Reply:
I don't think sphinx has the concept of a class vs an instance of a class. The `:class:` is used to cross-reference to the documentation for a class and I think they leave it up to the user as to what other words they want to put around it. The sphinx docs also say `:obj:` is to "Reference an object of unspecified type.". I don't know if it makes a difference in the rendered docs. 

I find the typing easier to reason if return types are used (`BaseScmRun` vs `Type[BaseScmRun]`). We have to wait until 3.10 to be mainstream for https://peps.python.org/pep-0673/ which will make typing these self returning classes more intuative.

The <> syntax describes the full name of the object which is used to cross-reference the docs. The text outside of the <> is what is link text that the user sees. Sphinx often can't find the cross-reference if the full import path isn't provided. Looking at the docs again this could be changed to `:class:`~scmdata.run.BaseScmRun``

```If you prefix the content with ~, the link text will only be the last component of the target. For example, :py:meth:`~Queue.Queue.get` will refer to Queue.Queue.get but only display get as the link text.```

Reply:
> I don't know if it makes a difference in the rendered docs.

I had a play, I don't think so. So it seems it's up to us. I guess you'd almost never returned an uninitialised class so probably doesn't matter/is obvious in context

> I find the typing easier to reason if return types are used (BaseScmRun vs Type[BaseScmRun])

Did I bring this up or is this a separate conversation?

Thanks for the info re <> syntax

Reply:
Re intersphinx for xarray in case it's helpful

```
 
 from importlib.metadata import version
 
+# see https://github.com/pydata/xarray/issues/4279
+import xarray
+
+xarray.DataArray.__module__ = "xarray"
+xarray.Dataset.__module__ = "xarray"
+
 # -- Project information -----------------------------------------------------
 # https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
 
@@ -65,6 +71,7 @@ intersphinx_mapping = {
     "python": ("https://docs.python.org/3", None),
     "pyam": ("https://pyam-iamc.readthedocs.io/en/latest", None),
     "scmdata": ("https://scmdata.readthedocs.io/en/latest", None),
+    "xarray": ("http://xarray.pydata.org/en/stable/", None),
     # "pint": ("https://pint.readthedocs.io/en/latest", None), # no full API doc here, unfortunately
 }
```

Reply:
> Did I bring this up or is this a separate conversation?
Just a brain dump on the comparative benefits of sphinx types vs python annotations because I had a play during this MR 😁

Reply:
Ah ok, you can just use the self type as the return type but yes that bug discussed in https://peps.python.org/pep-0673/ is annoying